conn.go

```diff
@@ -360,7 +359,13 @@ func (c *conn) Start() error {
 func (c *conn) Close() error {
 	c.closeMuxOnce.Do(func() { close(c.closeMux) })
 	err := c.Err()
-	if err == ErrConnClosed {
+	var connErr *ConnectionError
+	if errors.As(err, &connErr) && connErr.inner == nil {
+		// an empty ConnectionError means the connection was closed by the caller
+		// or as requested by the peer and no error was provided in the close frame.
+		// TODO: unclear if returning an error here is valuable.  it's not really
```

Comment:
This is a great question.

It seems that if the link is closed, detached or the connection itself is closed, that it'd be fine to return 'nil' here. Otherwise it seems like we failed to close it somehow and that's not the case.

Reply:
It's just like `io.Closer`.  The `Close` method returns an error but nobody checks it because how would you respond to it?  It's probably interesting from a logging/data collection standpoint, but that's about it.

Reply:
And to clarify, if `Close` is "clean", it will return nil.  It will only be non-nil if there's some pending error that hasn't been surfaced yet or closing the underlying network connection returns an error.

Reply:
OK one case where this is useful is if the peer closed the connection and sent back an error which presumably indicates why.

Reply:
I think I'll remove my comment block now that we've thought this through a bit.